src/components/GraphUtils.js

```diff
@@ -0,0 +1,202 @@
+import * as d3 from 'd3';
+
+export function truncate(fullStr, strLen, separator='...') {
+  if (fullStr.length <= strLen) return fullStr;
+  
+  const sepLen = separator.length,
+    charsToShow = strLen - sepLen,
+    frontChars = Math.ceil(charsToShow / 2),
+    backChars = Math.floor(charsToShow / 2);
+  
+  return fullStr.substr(0, frontChars) +
+    separator +
+    fullStr.substr(fullStr.length - backChars);
+}
+
+export function clearClicked() {
+  d3.selectAll('.clicked').classed('clicked', false);
+}
+
+export function scaleGraph(svg, g, zoom) {
+  const width = svg.node().getBoundingClientRect().width;
+  const height = svg.node().getBoundingClientRect().height;
+
```

Comment:
I think we could split this module even further by separating functions related to visualization and functions related strictly to graph processing (filtering, traversal, etc.).

These functions could be extracted into visualization module (e.g., `Visualization.js`):

* `clearClicked`
* `scaleGraph`
* `drag`

Other functions could be extracted into the graph module (e.g., `GraphUtils.js`).